src/pre-commit.ts

```diff
@@ -0,0 +1,309 @@
+
+const commonRepoOptions: Fig.Option[] = [
+  {
+    name: ["--verbose", "-v"],
+  },
+  {
+    name: ["--all-files", "-a"],
+    description: "Run all files in the repo",
+  },
+  {
+    name: "--files",
+    description: "Specific filenames to run hooks on",
+    args: {
+      name: "FILES",
+      isVariadic: true,
+      template: ["filepaths", "folders"],
+    },
+  },
+  {
+    name: "--show-diff-on-failure",
+    description: "When hooks fail, run `git diff` directly afterward",
+  },
+  {
+    name: "--hook-stage",
+    description: "The stage during which the hook is fired",
+    args: {
+      suggestions: hookStages,
+    },
+  },
+  {
+    name: "--remote-branch",
+    description: "Remote branch ref used by `git push`",
+    args: {
+      name: "REMOTE_BRANCH",
+      // TODO: am I allowed to import a generator for `git.ts`?
+    },
+  },
+  {
+    name: "--local-branch",
+    description: "Local branch ref used by `git push`",
+    args: {
+      name: "LOCAL_BRANCH",
+    },
+  },
+  {
+    name: ["--from-ref", "--source", "-s"],
+    description:
+      "Represents the original ref in a `from_ref...to_ref` diff expression. For `pre-push` hooks, this represents the branch you are pushing to. For `post-checkout` hooks, this represents the branch that was previously checked out",
+    args: {
+      name: "FROM_REF",
+    },
+  },
+  {
+    name: ["--to-ref", "--origin", "-o"],
+    description:
+      "Represents the destination ref in a `from_ref...to_ref` diff expression. For `pre-push` hooks, this represents the branch being pushed. For `post-checkout` hooks, this represents the branch that is now checked out",
+    args: {
+      name: "TO_REF",
+    },
+  },
+  {
+    name: "--commit-msg-filename",
+    description: "Filename to check when running during `commit-msg`",
+    args: {
+      name: "COMMIT_MSG_FILENAME",
+    },
+  },
+  {
+    name: "--remote-name",
+    description: "Remote name used by `git push`",
+    args: {
+      name: "REMOTE_NAME",
+    },
+  },
+  {
+    name: "--remote-url",
+    description: "Remote URL used by `git push`",
+    args: {
+      name: "REMOTE_URL",
+    },
+  },
+  {
+    name: "--checkout-type",
+    description:
+      "Indicates whether the checkout was a branch checkout (changing branches, flag=1) or a file checkout (retrieving a file from the index, flag=0)",
+    args: {
+      name: "CHECKOUT_TYPE",
+    },
+  },
+  {
+    name: "--is-squash-merge",
+    description:
+      "During a post-merge hook, indicates whether the merge was a squash merge",
+    args: {
+      name: "IS_SQUASH_MERGE",
+    },
+  },
+  {
+    name: "--rewrite-command",
+    description:
+      "During a post-rewrite hook, specifies the command that invoked the rewrite",
+    args: {
+      name: "REWRITE_COMMAND",
+    },
+  },
+];
+
+const completionSpec: Fig.Spec = {
+  name: "pre-commit",
+  description: "Pre-commit",
+  args: {},
+  options: [
+    globalOptions[0], // top level only takes `help`
```

Comment:
I wasn't sure about this approach, but it works. Happy to take any advice on a better accessor for the `help` without having to split out the option from `glopbalOptions`

Reply:
Can't we just do this?
```suggestion
    help,
```

Reply:
It seems like `globalOptions` are _already_ split out. I would just use the existing `help` value to make this more explicit.

Reply:
Oh, that's sneaky. Will update!